src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,187 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private MyLinkedList.Node<T> first;
+    private MyLinkedList.Node<T> last;
+
+    public MyLinkedList() {
+        this.size = 0;
+        first = null;
+        last = null;
+    }
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            addFirst(value);
+        } else {
+            addLast(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (!(index >= 0 && index <= size)) {
+            throw new IndexOutOfBoundsException("Wrong index value");
```

Comment:
```suggestion
            throw new IndexOutOfBoundsException("Wrong index value. Index: " + index);
```

нужно делать сообщения исключений как можно более информативными

Reply:
Зрозуміло